backend/src/algorithm/utils/constraint.ts

```diff
@@ -0,0 +1,79 @@
+import { Class, MatrixHashmap, RemainingLectures, Subject, Unavailable } from '../types';
+import { getLectureCountForDay } from './subject';
+import { hash } from './hash';
+import { maxSubjectLecturesPerDay } from '../consts';
+
+function checkIfClassUnavailable(timetable: MatrixHashmap, class_: Class, dayIndex: number, periodIndex: number) {
+  return timetable[hash(class_.name)][dayIndex][periodIndex];
+}
+
+function checkIfTeacherUnavailable(unavailable: Unavailable, subject: Subject, dayIndex: number, periodIndex: number) {
+  if (subject === null) return;
+  return unavailable.teachers[hash(subject.teacher)][dayIndex][periodIndex];
+}
+
+function checkIfClassroomUnavailable(
+  unavailable: Unavailable,
+  subject: Subject,
+  dayIndex: number,
+  periodIndex: number,
+) {
+  if (subject === null) return;
+  return subject.classroom && unavailable.classrooms[hash(subject.classroom)][dayIndex][periodIndex];
+}
+
+function checkIfLectureQuantityFulfilled(remainingLectures: RemainingLectures, class_: Class, subject: Subject) {
+  if (subject === null) return;
+  return remainingLectures[class_.name][subject.name] === 0;
+}
+
+function checkIfDailyLimitExceeded(timetable: MatrixHashmap, class_: Class, subject: Subject, dayIndex: number) {
+  if (subject === null) return;
+  const lectureCount = getLectureCountForDay(timetable, class_, subject, dayIndex) as number;
+  return lectureCount >= maxSubjectLecturesPerDay;
+}
+
+export function checkConstraints(
```

Comment:
Nitpick: this name does not give a good enough sense of what it produces. Does it return true because all of the constraints are passed and the timetable did not break it? Or does it return true because it did break them>

Reply:
It returns true in case constraint is broken. Renamed it  👍